### DISCUSSION TITLE
Pressing i3 audio and brightness hot keys do not notify

### DISCUSSION THREAD

**Discussion by chlin501** [i3/i3#6301] [Q&A] [answered]
### Environment
* OS: Debian trixie
* i3: 4.23 (2023-10-29)
* dunst: 1.11.0 (2024-04-15)
* i3status:  2.14-non-git
* xfce4 power manager: 4.18.4
### Problem
When pressing audio and brightness hot key, dunst does not show any notification message. Dunst is still working without a problem, because Gmail still notifies through dunst when a new email comes in, and other notification messages.  

My i3 config is as below. 
```
# Use pactl to adjust volume in PulseAudio.
set $refresh_i3status killall -SIGUSR1 i3status
bindsym XF86AudioRaiseVolume exec --no-startup-id pactl set-sink-volume @DEFAULT_SINK@ +10% && $refresh_i3status
bindsym XF86AudioLowerVolume exec --no-startup-id pactl set-sink-volume @DEFAULT_SINK@ -10% && $refresh_i3status
bindsym XF86AudioMute exec --no-startup-id pactl set-sink-mute @DEFAULT_SINK@ toggle && $refresh_i3status
bindsym XF86AudioMicMute exec --no-startup-id pactl set-source-mute @DEFAULT_SOURCE@ toggle && $refresh_i3status
# bindsym XF86MonBrightnessDown exec --no-startup-id brightnessctl --min-val=2 -q set 3%-
# bindsym XF86MonBrightnessUp exec --no-startup-id brightnessctl -q set 3%+
```
XF86MonBrightness{Down/Up} is commented out, but this is because I notice it's controlled by xfce power manager's `Handle display brightness keys`. If that setting is turned off, and uncomment XF86MonBrightness{Down/Up} part (then logout and login or reboot again), no dunst notification will appear on the top right corner. 

Also, I am sure **notify-send** is working without a problem. For instance, `notify-send mytitle mymessage`  shows dunst message popping at the top right corner with title showing mytitle, and message mymessage. 

Any places I should check for troubleshooting? Thanks

**Comment by okraits** [comment]
The keybindings in the i3 config only change the the volume or brightness level and trigger a refresh of i3status. They do not call `notify-send`. Why should a notification appear then?

**Reply by chlin501** [reply]
That's weird. I did fresh install Debian testing with i3, and the notification was working. It's like https://imgur.com/Y0gi6wb where the notification will pop up while pressing the XF86AudioRaiseVolume hot key. 

May I ask which part in the code or scripts used to generate config after the first time login (saved to .config/i3/config)? Maybe I can check which parts I changed that cause the issue.  Thanks.

**Reply by chlin501** [reply]
Never mind. I will try checking by adding notify-send instead. Thanks.

**Reply by orestisfl** [reply, marked as answer]
I have something in my dotfiles if it's helpful:
https://github.com/orestisfl/dotfiles/blob/b85fdc7fd589d419c32a01d17e31e22d5fd98b8b/i3/.config/i3/include/bindsym.config#L13-L15
https://github.com/orestisfl/dotfiles/blob/b85fdc7fd589d419c32a01d17e31e22d5fd98b8b/executables/bin/dunstify-volume.sh

**Reply by chlin501** [reply]
Thanks. That's useful!
